File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/activity/base/GGHFSRVModelActivity.java
package io.ganguo.demo.view.activity.base;

import io.ganguo.viewmodel.common.HFSRecyclerViewModel;
import io.ganguo.viewmodel.common.HeaderItemViewModel;
import io.ganguo.viewmodel.common.HeaderViewModel;
import io.ganguo.viewmodel.databinding.IncludeHfSwipeRecyclerBinding;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.vmodel.ViewModelHelper;

/**
 * HFSRecyclerViewModel - Activity
 * Created by leo on 2018/7/30.
 */
public abstract class GGHFSRVModelActivity<T extends IncludeHfSwipeRecyclerBinding, B extends HFSRecyclerViewModel> extends GGVModelActivity<T, B> {

    @Override
    public void onViewAttached(B viewModel) {
        ViewModelHelper.bind(viewModel.getHeaderContainer(), viewModel, onCreateHeaderVModel());
    }

}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/activity/base/GGHFSRVModelActivity.kt
package io.ganguo.demo.view.activity.base

import io.ganguo.viewmodel.core.ViewModelHelper
import io.ganguo.viewmodel.databinding.IncludeHfSwipeRecyclerBinding
import io.ganguo.viewmodel.pack.common.HFSRecyclerViewModel

/**
 * HFSRecyclerViewModel - Activity
 * Created by leo on 2018/7/30.
 */
abstract class GGHFSRVModelActivity<T : IncludeHfSwipeRecyclerBinding, B : HFSRecyclerViewModel<*>> : GGVModelActivity<T, B>() {


    override fun onViewAttached(viewModel: B) {
        ViewModelHelper.bind(viewModel.headerContainer, viewModel, newHeaderVModel())
    }
}
